### PR TITLE
Integrate explainability for hybrid query into RRF processor

### PR DESCRIPTION
**Pull request by martin-gaievski**
### Description
Adding explainability support to RRF processor

This PR adds explainability functionality to the RRF (Reciprocal Rank Fusion) processor. It's being submitted to the feature branch while the application security review for the RRF feature is in progress.

Key changes:
- Implemented explainability support for RRF
- Changed explanation details for case when sorting is enabled by field other then `score`, we return `0.0` as the score in explanation
- Minor refactoring and more unit tests

PR for explainability feature https://github.com/opensearch-project/neural-search/pull/1014

Example response when RRF is part of the search pipeline and the 'explain' flag is set:
```
"hits": {
        "total": {
            "value": 5,
            "relation": "eq"
        },
        "max_score": 0.032522473,
        "hits": [
            {
                "_shard": "[index-test][0]",
                "_node": "oEdoEKSrReuk_oX9PJilBg",
                "_index": "index-test",
                "_id": "LJS935MBlqE6ecphe2i6",
                "_score": 0.032522473,
                "_source": {
                    "field1": 50,
                    "vector": [
                        4.2,
                        5.5,
                        8.9
                    ],
                    "name": "Why would he go to all that effort for a free pack of ranch dressing?",
                    "category": "story",
                    "price": 10
                },
                "_explanation": {
                    "value": 0.032522473,
                    "description": "rrf combination of:",
                    "details": [
                        {
                            "value": 0.016393442,
                            "description": "rrf, rank_constant [60] normalization of:",
                            "details": [
                                {
                                    "value": 1.0,
                                    "description": "field1:[20 TO 150]",
                                    "details": []
                                }
                            ]
                        },
                        {
                            "value": 0.016129032,
                            "description": "rrf, rank_constant [60] normalization of:",
                            "details": [
                                {
                                    "value": 0.019948136,
                                    "description": "within top 12",
                                    "details": []
                                }
                            ]
                        }
                    ]
                }
            },
```

### Check List
- [X] New functionality includes testing.
- ~~[ ] New functionality has been documented.~~
- ~~[ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).~~
- [X] Commits are signed per the DCO using `--signoff`.
- ~~[ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).~~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/neural-search/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/neuralsearch/processor/ExplanationResponseProcessor.java [hunk 1]
```diff
@@ -111,8 +111,9 @@ public SearchResponse processResponse(
                         );
                     }
                     // Create and set final explanation combining all components
+                    Float finalScore = Float.isNaN(searchHit.getScore()) ? 0.0f : searchHit.getScore();
```
Comment (martin-gaievski) [inline src/main/java/org/opensearch/neuralsearch/processor/ExplanationResponseProcessor.java:114]: When sorting by a field other than `score`:
1. `searchHit.score` will be `Float.NaN`
2. `score` in the search hit and in actual response will be `null`

We can't pass `null` as a value for the explanation object. Therefore, I've set it to `0.0` in these cases. This ensures that we always have a valid numeric value for the explanation, even when the score isn't the primary sorting factor.

File: src/main/java/org/opensearch/neuralsearch/processor/combination/RRFScoreCombinationTechnique.java [hunk 2]
```diff
@@ -6,27 +6,39 @@
...
  * Abstracts combination of scores based on reciprocal rank fusion algorithm
  */
 @ToString(onlyExplicitlyIncluded = true)
-public class RRFScoreCombinationTechnique implements ScoreCombinationTechnique {
+public class RRFScoreCombinationTechnique implements ScoreCombinationTechnique, ExplainableTechnique {
     @ToString.Include
     public static final String TECHNIQUE_NAME = "rrf";
 
     // Not currently using weights for RRF, no need to modify or verify these params
-    public RRFScoreCombinationTechnique(final Map<String, Object> params, final ScoreCombinationUtil combinationUtil) {}
```
Comment (martin-gaievski) [inline src/main/java/org/opensearch/neuralsearch/processor/combination/RRFScoreCombinationTechnique.java:22]: those parameters were never used

**Review by ryanbogan — APPROVED** [review]
LGTM!

**Comment by heemin32** [comment]
Should we add more unit test to increase test coverage?
`Patch coverage is 62.50000% with 12 lines`

**Comment by martin-gaievski** [comment]
> Should we add more unit test to increase test coverage? `Patch coverage is 62.50000% with 12 lines`

Actual coverage should be higher than that. I've added several unit tests, and the run that is currently in this PR, is old (12/18 ~2pm PST). All next runs got throttled because of the shared limited quota for uploads ([example](https://github.com/opensearch-project/neural-search/actions/runs/12418190347/job/34670858726?pr=1037#step:7:34))

**Pull request merged**